compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3466,6 +3466,28 @@ boolean checkLiteralAssignabilityBasedOnType(BLangLiteral baseLiteral, BLangLite
         return false;
     }
 
+    boolean isFloatLiteralValue(Location pos, String numericLiteral) {
```

Comment:
This isn't simply a boolean check?

```suggestion
    BType validateFloatLiteral(Location pos, String numericLiteral) {
```